project/src/main/webapp/events-feed.js

```diff
@@ -23,6 +23,29 @@ $(async function() {
   }
 });
 
+/**
+ * Get all search events for the keyword entered and populate
+ * the events in the display.
+ */
+async function getAllSearchEvents() {
+  const eventContainer = document.getElementById('event-container');
+  if (eventContainer != null) {
+    eventContainer.remove();
+  }
+
+  const searchContainer = document.getElementById('search-container');
+  searchContainer.innerHTML = '';
+
+  const keyword = document.getElementById('keyword').value;
+  const response = await fetch(`search-data?keyword=${keyword}`);
+  const searchEvents = await response.json();
+  for (const key in searchEvents) {
+    if (searchEvents.hasOwnProperty(key)) {
+      populateEventContainer(searchEvents[key], 'search-container', 4);
```

Comment:
What's the difference between the old level 3 and this new level 4?

Otherwise, LGTM

Reply:
In level 4, the cards are placed into strict rows to demonstrate search ranking rather than the puzzling effect.